compiler/aarch64/codegen/ARM64BinaryEncoding.cpp

```diff
@@ -95,6 +95,12 @@ uint8_t *TR::ARM64ImmSymInstruction::generateBinaryEncoding()
       else
          {
          TR::MethodSymbol *method = symRef->getSymbol()->getMethodSymbol();
+
```

Comment:
I think this check needs to be sooner--before the `isRecursiveMethodTarget` check.  Otherwise, you'll miss trampolines created for recursive calls.

Reply:
I placed the check here because the recursive call case and the label case don't use trampolines.

Reply:
The recursive call case should use trampolines.  In a recompilation scenario, if the method is recompiled there could be a call from the old body to the new body (i.e., a recursive call).  The new body may require a trampoline to reach from the old body.

If we don't create a trampoline for this case already, that's probably another potential bug lurking.

Reply:
Since recompilation is not implemented yet for AArch64 this is not a real concern yet.  However, since you've opened #5059 to track I'm OK merging as-is.